cmd/solution/delete.go

```diff
@@ -0,0 +1,196 @@
+
+	solutionDeleteCmd.Flags().
+		Bool("yes", false, "Skip warning message and bypass confirmation step")
+
+	solutionDeleteCmd.MarkFlagsMutuallyExclusive("wait", "no-wait")
+
+	return solutionDeleteCmd
+}
+
+func deleteSolution(cmd *cobra.Command, args []string) {
+	var confirmationAnswer string
+	var solutionName string
+	var solutionTag string
+	var existingSolutionDeletionObjectId string
+
+	solutionTag, _ = cmd.Flags().GetString("tag")
+	skipConfirmationMessage, _ := cmd.Flags().GetBool("yes")
+	waitForDeletionDuration, _ := cmd.Flags().GetInt("wait")
+	noWait, _ := cmd.Flags().GetBool("no-wait")
+
+	solutionName = getSolutionNameFromArgs(cmd, args, "")
+
+	headers := map[string]string{
+		"tag": solutionTag,
+	}
+
+	if !skipConfirmationMessage {
+		fmt.Printf("WARNING! This command will remove all of the objects and types that are associated with this solution and will purge all data related to those objects and types.  It will also remove all solution metadata (including, but not limited to, subscriptions and other related objects).\nProceed with caution!  \nPlease type the name of the solution you want to delete and hit enter confirm that you want to delete the solution with name: %s and tag: %s \n", solutionName, solutionTag)
+		fmt.Scanln(&confirmationAnswer)
+
+		if confirmationAnswer != solutionName {
+			log.Fatal("Solution delete not confirmed, exiting command")
+		}
+	}
+
+	existingDeletionObj := getSolutionDeletionObject(solutionTag, solutionName)
+
+	if !existingDeletionObj.IsEmpty() {
+		existingSolutionDeletionObjectId = existingDeletionObj.ID
+	}
+
+	solutionDeleteUrl := fmt.Sprintf(getSolutionDeleteUrl(), solutionName)
+
+	var res any
+	err := api.JSONDelete(solutionDeleteUrl, &res, &api.Options{Headers: headers})
+	if err != nil {
+		log.Fatalf("Solution delete command failed: %v", err)
+	}
+
+	if !noWait {
+		output.PrintCmdStatus(cmd, fmt.Sprintf("Solution deletion initiated for solution with name: %s and tag: %s\n", solutionName, solutionTag))
+		var deletionObjData SolutionDeletionData
+		var newDeletionObjectId string
+		waitStartTime := time.Now()
+
+		for deletionObjData.Status == "" || deletionObjData.Status == "inProgress" || deletionObjData.IsEmpty() || newDeletionObjectId == existingSolutionDeletionObjectId {
+			output.PrintCmdStatus(cmd, fmt.Sprintf("Waited %f seconds for solution with name: %s and tag: %s to be marked as deleted\n", time.Since(waitStartTime).Seconds(), solutionName, solutionTag))
+			if time.Since(waitStartTime).Seconds() > float64(waitForDeletionDuration) {
+				log.Fatalf("Failed to validate solution with name %s and tag: %s was deleted: timed out", solutionName, solutionTag)
+			}
+			deletionObj := getSolutionDeletionObject(solutionTag, solutionName)
+			deletionObjData = deletionObj.DeletionData
+			newDeletionObjectId = deletionObj.ID
+			log.Infof("Got value of new deletion objectID :%s", newDeletionObjectId)
```

Comment:
```suggestion
```
Do you still need this? OK if yes, just let's use WithField or similar and reflect not only the objectID but maybe the status

Reply:
Resolved